### PR TITLE
Tree view filtering + load more

### PR DESCRIPTION
**Pull request by nl0**
## TODO

- [x] UI polish

### CODE REVIEW

**Comment by nl0** [comment]
@akarve 

> This is only for immediate children of current directory, right?

right

> It does complete substring matching, right?

right

> What happens to folders with thousands of children (e.g. we cannot fetch all candidates in a single request?) - we need to handle this case because some users have gigantic folders and want to cut down to a few files.

discussed in slack -- i'll add a "load more" button

> Filter placeholder text should be descriptive like "Filter children by prefix" or whatever.

"Filter children by substring"? or maybe you can come up with some better wording?

File: catalog/app/containers/Bucket/Listing.js [hunk 1]
```diff
@@ -85,149 +73,265 @@ const computeStats = R.reduce(
...
+                      <M.Icon
+                        style={{
+                          fontSize: 16,
+                          verticalAlign: 'text-bottom',
+                          marginRight: 4,
+                        }}
+                      >
+                        warning
+                      </M.Icon>
+                      Listing truncated to {totalItems} items
```
Comment (akarve) [inline catalog/app/containers/Bucket/Listing.js]:
```suggestion
                      Results truncated to {totalItems} items
```

File: catalog/app/containers/Bucket/Listing.js [hunk 2]
```diff
@@ -85,149 +73,265 @@ const computeStats = R.reduce(
...
-    )
+    [filtering.set, inputRef],
+  )
+
+  return (
+    <div className={classes.root}>
+      <M.InputBase
+        {...filtering.input}
+        onKeyDown={handleKeyDown}
+        placeholder="Filter children of current directory"
```
Comment (akarve) [inline catalog/app/containers/Bucket/Listing.js]:
```suggestion
        placeholder="Filter current directory by substring"
```

Comment (akarve) [reply catalog/app/containers/Bucket/Listing.js]: Wordsmith filtering text

**Pull request merged**